src/oom/oomd-util.c

```diff
@@ -278,6 +284,25 @@ int oomd_cgroup_context_acquire(const char *path, OomdCGroupContext **ret) {
         if (r < 0)
                 return log_debug_errno(r, "Error parsing memory pressure from %s: %m", p);
 
+        r = cg_get_owner(SYSTEMD_CGROUP_CONTROLLER, path, &uid, &gid);
+        if (r < 0)
+                log_debug_errno(r, "Failed to get owner/group from %s: %m", path);
+        else if (uid == 0 && gid == 0) {
+                /* Ignore most errors when reading the xattr since it is usually unset and cgroup xattrs are only used
+                 * as an optional feature of systemd-oomd (and the system might not even support them). */
+                r = cg_get_xattr_malloc(SYSTEMD_CGROUP_CONTROLLER, path, "user.oomd_avoid", &avoid_val);
+                if (r >= 0 && streq(avoid_val, "1"))
+                        ctx->avoid = true;
+                else if (r == -ENOMEM)
+                        return r;
+
+                r = cg_get_xattr_malloc(SYSTEMD_CGROUP_CONTROLLER, path, "user.oomd_omit", &omit_val);
+                if (r >= 0 && streq(omit_val, "1"))
+                        ctx->omit = true;
+                else if (r == -ENOMEM)
+                        return r;
```

Comment:
hmm, why is this a single enum in pid1 btw, and here two bools? why not make it an enum all the way, i.e. in pid1, in the xattr and in oomd, too?